addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -45,9 +45,10 @@ const FormEditor = options.Class.extend({
                 method: 'search_read',
                 args: [
                     field.domain,
-                    ['display_name']
+                    field.name_field || ['display_name']
                 ],
             });
+            field.records.forEach(r => r.display_name = r[field.name_field]);
```

Comment:
Maybe just add a comment to say we do this for the renderer to display that value, since we have no control over it. (Or something more concise :p)

A bit sad we have to do this but short of using a custom template for rendering the fields I don't see a solution.
I think there could be more use cases for stuff like this so it should be ok.

We could do the trick with display_name but I still prefer this actually.. we'll see what the reviewer says I guess :)